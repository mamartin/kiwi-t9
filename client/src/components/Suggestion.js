// @flow
import React from "react"
import { TouchableOpacity, Text } from "react-native"

const styles = {
  suggestionWrapper: {
    display: "flex",
    marginRight: 30,
    height: 60,
    justifyContent: "center",
  },
  text: {
    color: "white",
    fontSize: 30,
  },
}

type SuggestionProps = {|
  word: string,
  onPress: (?string) => null,
|}

export default class Suggestion extends React.PureComponent<SuggestionProps> {
  render() {
    const { word, onPress } = this.props
    return (
      <TouchableOpacity
        onPress={() => {
          if (typeof onPress === "function") {
            onPress(word)
          }
        }}
        style={styles.suggestionWrapper}
      >
        <Text style={styles.text}>{word}</Text>
      </TouchableOpacity>
    )
  }
}
