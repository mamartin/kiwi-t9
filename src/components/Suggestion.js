// @flow
import React from "react"
import { View, Text } from "react-native"

const styles = {
  suggestionWrapper: {
    paddingRight: 30,
  },
  text: {
    fontSize: 30,
  },
}

type SuggestionProps = {|
  word: string,
|}

export default class Suggestion extends React.PureComponent<SuggestionProps> {
  render() {
    const { word } = this.props
    return (
      <View style={styles.suggestionWrapper}>
        <Text style={styles.text}>{word}</Text>
      </View>
    )
  }
}
