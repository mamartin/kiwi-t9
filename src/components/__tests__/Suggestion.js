import React from "react"
import renderer from "react-test-renderer"

// components
import Suggestion from "../Suggestion"

describe("Suggestion component", () => {
  it("renders with default props", () => {
    const tree = renderer.create(<Suggestion word="Hello" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
