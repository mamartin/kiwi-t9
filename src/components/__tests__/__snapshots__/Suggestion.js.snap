// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestion component renders with default props 1`] = `
<View
  style={
    Object {
      "paddingRight": 30,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 30,
      }
    }
  >
    Hello
  </Text>
</View>
`;
